nuxeo-management-rest-api/src/test/java/org/nuxeo/rest/management/TestElasticsearchObject.java

```diff
@@ -0,0 +1,202 @@
+import org.nuxeo.elasticsearch.api.ElasticSearchAdmin;
+import org.nuxeo.elasticsearch.api.ElasticSearchService;
+import org.nuxeo.elasticsearch.query.NxQueryBuilder;
+import org.nuxeo.jaxrs.test.CloseableClientResponse;
+import org.nuxeo.runtime.management.ManagementFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features(ManagementFeature.class)
+@Deploy("org.nuxeo.elasticsearch.core")
+@Deploy("org.nuxeo.elasticsearch.core.test:elasticsearch-test-contrib.xml")
+@Deploy("org.nuxeo.ecm.core.management")
+public class TestElasticsearchObject extends ManagementBaseTest {
+
+    public static final String GET_DOCUMENTS_QUERY = "SELECT * from Document";
+
+    public static final String ES_INDEXING_PATH = "site/management/elasticsearch/%s/reindex";
+
+    @Inject
+    protected CoreSession coreSession;
+
+    @Inject
+    protected ElasticSearchService ess;
+
+    @Inject
+    protected ElasticSearchAdmin esa;
+
+    @Inject
+    protected TransactionalFeature txFeature;
+
+    @Test
+    public void shouldRunIndexing() throws IOException {
+        // Init indexes and drop if any
+        esa.initIndexes(true);
+
+        // Initial docs count
+        long initialDocCount = coreSession.query(GET_DOCUMENTS_QUERY).totalSize();
+
+        // Nothing indexed because the index was dropped
+        assertEquals(0, ess.query(new NxQueryBuilder(coreSession).nxql(GET_DOCUMENTS_QUERY)).totalSize());
+
+        // Create new documents without indexing them
+        createDocuments();
+
+        // Wait for an eventual indexing
+        waitForIndexing();
+
+        // Nothing indexed because of disable indexing flag
+        assertEquals(0, ess.query(new NxQueryBuilder(coreSession).nxql(GET_DOCUMENTS_QUERY)).totalSize());
+
+        // Start the ES indexing all document of the coreSession repository
+        String path = String.format(ES_INDEXING_PATH, coreSession.getRepositoryName());
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            verifyIndexingResponse(response, 3 + initialDocCount);
+        }
+    }
+
+    @Test
+    public void shouldRunIndexingByNXQLQuery() throws IOException {
+        // Init indexes and drop if any
+        esa.initIndexes(true);
+
+        // Create new documents without indexing them
+        createDocuments();
+
+        String path = String.format(ES_INDEXING_PATH, coreSession.getRepositoryName())
+                + "?query=Select * From Document where dc:title like 'Title of my-file%'";
+
+        // Start the ES indexing document that match the nxql query (2 files)
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            verifyIndexingResponse(response, 2);
```

Comment:
I wonder why you don't need to add the `initialDocCount` to the expected count as in the first test.

Reply:
yes, it's because for each test i init/drop the indexes, and in this test i index only the document that match the query (2 files only) it's why i give only 2 as expected.